### DISCUSSION TITLE
Control legend orientation

### DISCUSSION THREAD

**Discussion by lstillwe** [TerriaJS/terriajs#6407] [Q&A]
Is there any way to control legend orientation (i.e. make horizontal) from a WMS layer served by GeoServer?
I have specified an url like this in the legends trait:

```
"legends": [
                        { 
"url": "https://hostname.org/geoserver/workspace/ows?service=WMS&request=GetLegendGraphic&LEGEND_OPTIONS=layout:horizontal&format=image%2Fpng&width=20&height=20&layer=layername"
                        }
 ]
```
However that only gives me a link to see the legend in a new page:

![Screen Shot 2022-07-08 at 5 58 36 PM](https://user-images.githubusercontent.com/1794116/178075568-e756571e-841a-44ec-884e-85f0aee07519.png)

Thanks very much!

**Comment by nf-s** [comment]
Hi @lstillwe 

You need to set the `"urlMimeType"` trait for any Legend `url` which don't have one of the following suffixes/extensions - `png|jpg|jpeg|gif|svg`.

I just realised this isn't mentioned in the docs!
I have added a PR to update them https://github.com/TerriaJS/terriajs/pull/6412

This should work for your config

```json
  "legends": [
    {
      "url": "https://hostname.org/geoserver/workspace/ows?service=WMS&request=GetLegendGraphic&LEGEND_OPTIONS=layout:horizontal&format=image%2Fpng&width=20&height=20&layer=layername",
      "urlMimeType": "image/png"
    }
  ]
```

Cheers,
Nick
